src/__tests__/utils/puzzles.test.ts

```diff
@@ -0,0 +1,29 @@
+import { checkPuzzleAnswer } from '../../utils/puzzles';
+import { Puzzle } from '@prisma/client';
+
+it('should compare the user submitted puzzle answer with the ans key in the puzzle (from the db)', () => {
+  const puzzle = {
+    id: 1,
+    name: 'fake1',
+    createdAt: new Date(Date.now()),
+    updatedAt: new Date(Date.now()),
+    difficulty: 'EASY',
+    content: [],
+    imageUrl: null,
+    exampleContent: [],
+    exampleImageUrl: null,
+    isGenerated: true,
+    question: 'What is 1+1?',
+    inputType: 'TEXT',
+    published: false,
+    variables: {
+      answer: '2'
+    },
+    puzzleType: {},
+    puzzleTypeId: 1,
+    puzzleInstances: []
+  } as Puzzle;
+
+  expect(checkPuzzleAnswer(puzzle, 1, '3')).toBeFalsy();
+  expect(checkPuzzleAnswer(puzzle, 1, '2')).toBeTruthy();
```

Comment:
This is alright, but here's a new concept. Since we are creating fake objects in order to run these tests, it's good practice to not make the tests depend too heavily on the fake data. In other words, instead of hardcoding the `'2'` on line 28, you could put `expect(checkPuzzleAnswer(puzzle, 1, puzzle.variables.answer)).toBeTruthy();`, which logically performs the same test, but now we can change the fake object's answer anytime and the test doesn't need to be updated.

Lmk if you want to talk about this further, but hopefully ti makes sense

Reply:
Also smart way to generate the false value that references the real answer too

Reply:
Switch `toBeTruthy` and `toBeFalsey` into `toBe(false)` and `toBe(true)`